src/Arcus.Observability.Telemetry.Serilog.Filters/TelemetryTypeFilter.cs

```diff
@@ -44,27 +48,36 @@ public static TelemetryTypeFilter On(TelemetryType telemetryType)
         /// </summary>
         /// <param name="telemetryType">Current telemetry type that should be filtered</param>
         /// <param name="isTrackingEnabled">Indication whether or not this telemetry type should be tracked</param>
+        /// <exception cref="ArgumentOutOfRangeException">Thrown when the <paramref name="telemetryType"/> is outside the bounds of the enumeration.</exception>
         public static TelemetryTypeFilter On(TelemetryType telemetryType, bool isTrackingEnabled)
         {
+            Guard.For(() => !Enum.IsDefined(typeof(TelemetryType), telemetryType), 
+                new ArgumentOutOfRangeException(nameof(telemetryType), telemetryType, "Requires a type of telemetry that's within the supported value range of the enumeration"));
+            
             return new TelemetryTypeFilter(telemetryType, isTrackingEnabled);
         }
 
         /// <inheritdoc />
         public bool IsEnabled(LogEvent logEvent)
         {
-            if (IsTrackingEnabled == true)
+            if (IsTrackingEnabled is true)
             {
                 return true;
             }
 
-            var telemetryType = logEvent.Properties.GetAsEnum<TelemetryType>(ContextProperties.General.TelemetryType);
-            if (telemetryType == null)
+            switch (TelemetryType)
```

Comment:
Can't we revert this change now that we have this?

`ContextProperties.General.TelemetryType`?

Reply:
No because the telemetry type is a property of the log entry type, not the Serilog log event.